velox/docs/functions/spark/string.rst

```diff
@@ -0,0 +1,87 @@
+    multiple code points into a single user-perceived *character*, the basic
+    unit of a writing system for a language, but the functions will treat each
+    code point as a separate unit.
+
+.. spark:function:: ascii(string) -> integer
+
+    Returns the numeric value of the first character of ``string``.
+
+.. spark:function:: chr(n) -> varchar
+
+    Returns the Unicode code point ``n`` as a single character string.
+
+.. spark:function:: contains(left, right) -> boolean
+
+    Returns a boolean. The value is True if right is found inside left.
+    Returns NULL if either input expression is NULL. Otherwise, returns False.
+    Both left or right must be of STRING. ::
+
+        SELECT contains('js SQL', 'js'); -- true
+        SELECT contains('js SQL', 'js'); -- false
+        SELECT contains('js SQL', null); -- NULL
+
+.. spark:function:: endsWith(left, right) -> boolean
+
+    Returns a boolean. The value is True if left ends with right.
+    Returns NULL if either input expression is NULL. Otherwise, returns False.
+    Both left or right must be of STRING. ::
+
+        SELECT endswith('js SQL', 'SQL'); -- true
+        SELECT endswith('js SQL', 'js'); -- false
+        SELECT endswith('js SQL', null); -- NULL
+
+.. spark:function:: instr(string, substring) -> integer
+
+    Returns the starting position of the first instance of ``substring`` in
+    ``string``. Positions start with ``1``. If not found, ``0`` is returned.
+
+.. spark:function:: length(string) -> integer
+
+    Returns the length of ``string`` in characters.
+
+.. spark:function:: split(string, delimiter) -> array(string)
+
+    Splits ``string`` on ``delimiter`` and returns an array. ::
+
+        SELECT split('oneAtwoBthreeC', '[ABC]'); -- ["one","two","three",""]
+
+.. spark:function:: split(string, delimiter, limit) -> array(string)
+
+    Splits ``string`` on ``delimiter`` and returns an array of size at most ``limit``. ::
+
+        SELECT split('oneAtwoBthreeC', '[ABC]', -1); -- ["one","two","three",""]
+        SELECT split('oneAtwoBthreeC', '[ABC]', 2); -- ["one","twoBthreeC"]
+
+.. spark:function:: startsWith(left, right) -> boolean
+
+    Returns a boolean. The value is True if left starts with right.
+    Returns NULL if either input expression is NULL. Otherwise, returns False.
+    Both left or right must be of STRING. ::
+
+        SELECT startswith('js SQL', 'js'); -- true
+        SELECT startswith('js SQL', 'SQL'); -- false
+        SELECT startswith('js SQL', null); -- NULL
+
+.. spark:function:: substring(string, start) -> varchar
+
+    Returns the rest of ``string`` from the starting position ``start``.
+    Positions start with ``1``. A negative starting position is interpreted
+    as being relative to the end of the string.
+
+.. spark:function:: substring(string, start, length) -> varchar
+
+    Returns a substring from ``string`` of length ``length`` from the starting
```

Comment:
Perhaps, clarify the behavior when length is zero, negative or very large.

Reply:
add example
        SELECT substring('Spark SQL', 5, 0); -- ""
        SELECT substring('Spark SQL', 5, -1); -- ""
        SELECT substring('Spark SQL', 5, 10000); -- "k SQL"